### ISSUE TITLE
new Check: NoWhitespaceBeforeCaseDefaultColon

### ISSUE THREAD

**Issue by wilcoln** [checkstyle/checkstyle#8186]
The idea of this check originates from https://github.com/checkstyle/checkstyle/issues/8045
This check should report whitespaces before colon token of switch cases and default.
```
cat Test.java
class Test {
	{
		switch(1) {
			case 1 : // expected violation
				break;
                        case 2:
                                break;
			default : // expected violation
				break;
		}
	}
}

$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <module name="TreeWalker">
       <module name="NoWhitespaceBeforeCaseDefaultColon"/>
    </module>
</module>
/var/tmp $ java $RUN_LOCALE -jar checkstyle-8.31-all.jar -c config.xml TestClass.java
Starting audit...
[ERROR] /var/tmp/Test.java:4: ':' is preceded with whitespace. [NoWhitespaceBeforeCaseDefaultColon]
[ERROR] /var/tmp/Test.java:6: ':' is preceded with whitespace. [NoWhitespaceBeforeCaseDefaultColon]
Audit done.
Checkstyle ends with 2 errors.
```

Rationale: 
The official [google java style guide](https://google.github.io/styleguide/javaguide.html#s4.6-whitespace) does not allow whitespaces before the colon of switch cases and default. Adding this check will improve checkstyle's support for google's java style guide.
Also, we can see in comments of https://github.com/checkstyle/checkstyle/issues/8045 that users are well receptive to the check.

**Comment by pbludov** [comment]
Also, no whitespace is expected in [LabeledStatements](https://docs.oracle.com/javase/specs/jls/se8/html/jls-14.html#jls-LabeledStatement):
```java
outer: for(;;) {
   for(;;) {
     break outer;
   }
}
```
If we come up with a good name, we can also handle them.

**Comment by wilcoln** [comment]
@pbludov, we handle this already.
See Issue https://github.com/checkstyle/checkstyle/issues/8124 and corresponding PR https://github.com/checkstyle/checkstyle/pull/8125

**Comment by pbludov** [comment]
> @pbludov, we handle this already.
> See Issue #8124 and corresponding PR #8125

Sorry, I missed it. Well, it looks weird: actually, it should be "no whitespace **after** labeled statement." @romani may be it is better to revert #8124 and add this to the new check?

**Comment by wilcoln** [comment]
@pbludov I agree with you that it seems weird, but I think the weirdness comes from the name `LABELED_STAT` which is given to a token that is actually just the colon ':' of a labeled statement.
I personally think a name like `LABEL_COLON` would have been more appropriate. But that's another matter.

What we wanted was a check that reports whitespace before the colon of a label statement.
That colon is nothing else than `LABELED_STAT` token. That's why it makes sens to add it to `NoWhitespaceBefore`

**Comment by wilcoln** [comment]
> ... the weirdness comes from the name `LABELED_STAT` which is given to a token that is actually just the colon ':' of a labeled statement.
>... I personally think a name like `LABEL_COLON` would have been more appropriate. But that's another matter.

colons of labels are not `COLON` token, but are `LABELED_STAT`.

    |      |--LABELED_STAT -> : [4:24]
    |      |  |--IDENT -> mylabel [4:16]
    |      |  |--SINGLE_LINE_COMMENT -> // [4:27]
    |      |  |  `--COMMENT_CONTENT ->  expected violation\n [4:29]

**Comment by AkMo3** [comment]
I am on it.

**Comment by rnveach** [comment]
Fix was merged
